recipe/patches/0002-Remove-nvidia-nccl-cu12-from-pyproject.toml.patch

```diff
@@ -0,0 +1,26 @@
+From b1fdd547abf446e9376fc743e5450205da081ecc Mon Sep 17 00:00:00 2001
+From: Hyunsu Cho <phcho@nvidia.com>
+Date: Mon, 26 Aug 2024 18:21:24 -0700
+Subject: [PATCH] Remove nvidia-nccl-cu12 from pyproject.toml
+
+---
+ python-package/pyproject.toml | 3 +--
+ 1 file changed, 1 insertion(+), 2 deletions(-)
+
+diff --git a/python-package/pyproject.toml b/python-package/pyproject.toml
+index 05bbe998c..184a8be6c 100644
+--- a/python-package/pyproject.toml
++++ b/python-package/pyproject.toml
+@@ -29,8 +29,7 @@ classifiers = [
+ ]
+ dependencies = [
+     "numpy",
+-    "scipy",
+-    "nvidia-nccl-cu12 ; platform_system == 'Linux' and platform_machine != 'aarch64'"
++    "scipy"
```

Comment:
As [trailing commas are valid in TOML]( https://github.com/toml-lang/toml/blob/dc5af251cd348ee053d9355359743767e3d0b123/toml.md?plain=1#L667C33-L668C55 ), think we can clean this patch up a bit to just remove a line

Would also suggest adopting this practice upstream to make patching like this easier. Additionally it will make upstream PR diffs cleaner

```suggestion
+    "scipy",
-    "nvidia-nccl-cu12 ; platform_system == 'Linux' and platform_machine != 'aarch64'"
```

Reply:
Suggested a similar approach upstream: https://github.com/dmlc/xgboost/pull/10759